lib/tests/modules.sh

```diff
@@ -13,126 +16,121 @@ fail=0
 
 evalConfig() {
     local attr=$1
-    shift;
-    local script="import ./default.nix { modules = [ $@ ];}"
+    shift
+    local script="import ./default.nix { modules = [ $* ];}"
     nix-instantiate --timeout 1 -E "$script" -A "$attr" --eval-only --show-trace --read-write-mode
 }
 
 reportFailure() {
     local attr=$1
-    shift;
-    local script="import ./default.nix { modules = [ $@ ];}"
+    shift
+    local script="import ./default.nix { modules = [ $* ];}"
     echo 2>&1 "$ nix-instantiate -E '$script' -A '$attr' --eval-only"
-    evalConfig "$attr" "$@"
-    fail=$((fail + 1))
+    evalConfig "$attr" "$@" || true
```

Comment:
Does the function still abort if we catch the error here like it did before?

Reply:
> The original code didn't have errexit

Did you only check the bash file? Maybe nix stdenv or so set it before running the script? In normal packages the build will fail if any command errors.

I hope that the tests failed before if anything went wrong and they should fail the same way after this change. Could you try to invalidate a test and try running them to see if everything still works?

Reply:
> Did you only check the bash file? Maybe nix stdenv or so set it before running the script?

It's probably good not to rely on state outside of this script. But also the script has a shebang line, indicating that it should be run in its own interpreter instance.

Reply:
> I hope that the tests failed before if anything went wrong and they should fail the same way after this change. Could you try to invalidate a test and try running them to see if everything still works?

I've checked a few modifications, including making sure more than one test can fail without exiting the script:

```
$ git diff
diff --git a/lib/tests/modules.sh b/lib/tests/modules.sh
index 590937da5b8..1bb6d218b30 100755
--- a/lib/tests/modules.sh
+++ b/lib/tests/modules.sh
@@ -114,7 +114,7 @@ checkConfigError 'attribute .*foo.* .* not found' "$@" ./define-if-attrsOfSub-fo
 checkConfigError 'attribute .*foo.* .* not found' "$@" ./define-attrsOfSub-if-foo-enable.nix
 checkConfigError 'attribute .*foo.* .* not found' "$@" ./define-attrsOfSub-foo-if-enable.nix
 checkConfigOutput '^false$' "$@" ./define-attrsOfSub-foo-enable-if.nix
-checkConfigOutput '^true$' "$@" ./define-enable.nix ./define-if-attrsOfSub-foo-enable.nix
+checkConfigOutput '^tr ue$' "$@" ./define-enable.nix ./define-if-attrsOfSub-foo-enable.nix
 checkConfigOutput '^true$' "$@" ./define-enable.nix ./define-attrsOfSub-if-foo-enable.nix
 checkConfigOutput '^true$' "$@" ./define-enable.nix ./define-attrsOfSub-foo-if-enable.nix
 checkConfigOutput '^true$' "$@" ./define-enable.nix ./define-attrsOfSub-foo-enable-if.nix
@@ -124,7 +124,7 @@ set -- config.enable ./define-enable.nix ./declare-enable.nix
 checkConfigOutput '^true$' "$@"
 checkConfigOutput '^false$' "$@" ./disable-define-enable.nix
 checkConfigError "The option .*enable.* does not exist. Definition values:\n\s*- In .*: true" "$@" ./disable-declare-enable.nix
-checkConfigError "attribute .*enable.* in selection path .*config.enable.* not found" "$@" ./disable-define-enable.nix ./disable-declare-enable.nix
+checkConfigError "attrib ute .*enable.* in selection path .*config.enable.* not found" "$@" ./disable-define-enable.nix ./disable-declare-enable.nix
 checkConfigError "attribute .*enable.* in selection path .*config.enable.* not found" "$@" ./disable-enable-modules.nix
 
 # Check _module.args.
$ nix-build lib/tests/release.nix
these derivations will be built:
  /nix/store/bwl37hdqq10hg3fy57vgv4777by59jrp-nixpkgs-lib-tests.drv
building '/nix/store/bwl37hdqq10hg3fy57vgv4777by59jrp-nixpkgs-lib-tests.drv'...
Running lib/tests/modules.sh
error: Expected result matching '^tr ue$', while evaluating
$ nix-instantiate -E 'import ./default.nix { modules = [ ./declare-enable.nix ./declare-attrsOfSub-any-enable.nix ./define-enable.nix ./define-if-attrsOfSub-foo-enable.nix ];}' -A 'config.attrsOfSub.foo.enable' --eval-only
true
error: Expected error matching 'attrib ute .*enable.* in selection path .*config.enable.* not found', while evaluating
$ nix-instantiate -E 'import ./default.nix { modules = [ ./define-enable.nix ./declare-enable.nix ./disable-define-enable.nix ./disable-declare-enable.nix ];}' -A 'config.enable' --eval-only
error: attribute 'enable' in selection path 'config.enable' not found
====== module tests ======
130 Pass
2 Fail
builder for '/nix/store/bwl37hdqq10hg3fy57vgv4777by59jrp-nixpkgs-lib-tests.drv' failed with exit code 1
error: build of '/nix/store/bwl37hdqq10hg3fy57vgv4777by59jrp-nixpkgs-lib-tests.drv' failed
```